tf2onnx/shape_inference.py

```diff
@@ -403,3 +434,43 @@ def _find_tensorarray_write(graph, node):
                 if j.type == "TensorArrayWriteV3":
                     return j
     return None
+
+
+def _broadcasting_rule_check(shape_a, shape_b):
+    utils.make_sure(len(shape_a) == len(shape_b), "the following code logic assumes length is equal")
+
+    legal_flag = True
+    for i, j in zip(shape_a, shape_b):
+        # if dim value is -1 then we assume legal
+        # because if not then the model runner such as onnxruntime will issue an error
+        if 1 in [i, j] or -1 in [i, j]:
+            continue
+        if i != j:
+            legal_flag = False
+
+    return legal_flag
+
+
+def _shape_after_broadcasting(shape_a, shape_b):
+    # expand shape data according to broadcasting rule to make them have same length

```

Comment:
can this method meet your needs?
https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/shape_inference.py#L355